src/core/qgsvectorlayercache.cpp

```diff
@@ -231,6 +237,7 @@ void QgsVectorLayerCache::attributeAdded( int field )
 {
   Q_UNUSED( field )
   mCachedAttributes.append( field );
+  mFullCache = false;
```

Comment:
Thanks - fixed. I was under the mistaken impression that this was used to indicate that we'd filled the cache with all available features.